README.md

```diff
@@ -37,10 +38,14 @@ npm i @autotelic/fastify-hmac
 module.exports = function (fastify, options, next) {
   fastify.register(require('fastify-hmac'), {
     sharedSecret: 'topSecret',
-    getAlgorithm: () => 'sha512',
+    algorithmMap: {
+      hs2019: {
+        'test-key-a': 'sha512',
+        'test-key-b': 'sha256'
+      }
+    },
```

Comment:
This mapping pattern should work for the implementation we're currently using elsewhere (as James described it to me) but I'm not sure it is an example of the 'default' case as described by the [draft specificaion](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-00#section-4.1).

The `algorithm` signature property is indicated as "recommended" while the `keyId` property is indicated as "required". i.e. some 'default' implementations will need to determine an algorithm with the `keyId` property alone. This 'default' case will not work with this implementation if I am interpreting it correctly. 

On the other hand, this is a draft specification and the [issues](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-00#appendix-B.1) listed in the document specifically address the ambiguity of the `keyId` and `algorithm` interaction. All of this may change when they issue a final version which we would need to update to anyway.

Reply:
This actually looks good to me. A couple of things

The "default" case is somewhat up to us, there is a lot of room for flexibility with the implementation of these guidelines, and we arent supporting everything off the bat. WHat we are doing though, its providing some sensible defaults for the plugin. `algorithm` is recommended and in our implementation thats what we are using. For anyone using _only_ the keyID they should be able to override the necessary bits to select the correct algorithm. For example, if I ignore the algorithm property, I can use the following config overrides to achieve what you are talking about.
// pseudo example
```js
algorithmMap: {
  testKeyA: 'sha512',
}

getAlgorithm(request, options) {
 // parseSignature is a method that extracts keyId from the signature
  const { keyID} = parseSignature(request.signature)
  const algorithm = options.algorithmMap[keyID]
  return algorithm
}
```

Reply:
Just remember that the "defaults" are not meant to be one size fits all and a case that will work for _everything_, we do it in a way that makes sense for us according to the spec, but if anyone wishes to change it they have the ability to do so.